src/snowcli/cli/object/commands.py

```diff
@@ -14,45 +15,54 @@
     context_settings=DEFAULT_CONTEXT_SETTINGS,
     help="Manages Snowflake objects like warehouses and stages",
 )
-app.add_typer(stage_app)  # type: ignore
+app.add_typer(stage_app)
+
+
+def _check_if_supported_object(value: str) -> ObjectNames:
+    if value not in OBJECT_TO_NAMES:
+        raise ClickException(f"Object of type {value} is not supported.")
+    return OBJECT_TO_NAMES[value]
+
 
 NameArgument = typer.Argument(None, help="Name of the object")
-ObjectArgument = typer.Argument(None, help="Type of object")
+ObjectArgument = typer.Argument(
+    None,
+    help="Type of object. For example table, procedure, streamlit.",
+    case_sensitive=False,
+    callback=_check_if_supported_object,
+)
 LikeOption = typer.Option(
     "%%",
     "--like",
     "-l",
-    help='Regular expression for filtering the functions by name. For example, `list --like "my%"` lists all functions in the **dev** (default) environment that begin with “my”.',
+    help='Regular expression for filtering the functions by name. For example, `list --like "my%"` lists '
+    "all functions in the **dev** (default) environment that begin with “my”.",
 )
 
 
-@app.command()
+@app.command("list")
 @with_output
 @global_options_with_connection
-def show(
-    object_type: ObjectType = ObjectArgument,
+def list_(
+    object_type=ObjectArgument,
     like: str = LikeOption,
     **options,
 ):
-    "Lists all available Snowflake objects of given type"
-    return QueryResult(ObjectManager().show(object_type, like))
+    """Lists all available Snowflake objects of given type."""
+    return QueryResult(ObjectManager().show(object_type.sf_plural_name, like))
 
 
 @app.command()
 @with_output
 @global_options_with_connection
-def drop(
-    object_type: ObjectType = ObjectArgument, object_name: str = NameArgument, **options
-):
-    "Drops Snowflake object of given name and type"
-    return QueryResult(ObjectManager().drop(object_type, object_name))
+def drop(object_type=ObjectArgument, object_name: str = NameArgument, **options):
+    """Drops Snowflake object of given name and type."""
+    return QueryResult(ObjectManager().drop(object_type.sf_name, object_name))
```

Comment:
Why do we decide which name is required outside of ObjectManager? Why do commands need to have this knowledge? IMO commands should just call show/drop/describe and manager should decide which name is needed.

Reply:
Changed